src/ansiblelint/rules/NoLogPasswordsRule.py

```diff
@@ -0,0 +1,35 @@
+
+from ansiblelint.rules import AnsibleLintRule
+from ansiblelint.utils import convert_to_boolean
+from typing import TYPE_CHECKING, Any, Dict, Union
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    from ansiblelint.file_utils import Lintable
+
+
+class NoLogPasswordsRule(AnsibleLintRule):
+    id = "no-log-password"
+    shortdesc = "password should not be logged."
+    description = (
+        "All the modules that take a password argument should fail "
+        "if no_log is not set or set to False in the task."
+    )
+    severity = 'LOW'
+    tags = ["passwords", "experimental"]
```

Comment:
Replace passwords with `security`, describes the category in a more generic way.

Reply:
IMHO a description  message like below would be less likely to confuse users
```
When passing password argument you should have no_log configured to a non False value to avoid accidental leaking of secrets.
```

Reply:
Done